physlr/physlr.py

```diff
@@ -1585,31 +1562,19 @@ def determine_molecules_process(u):
 
     def physlr_molecules(self):
         "Separate barcodes into molecules."
-        strategy_switcher = {
-            1: "\n\tStrategy: "
-               "Bi-connected components separation",
-            2: "\n\tStrategy: "
-               "K-clique community detection (after separating bi-connected components)",
-            3: "\n\tStrategy: "
-               "Louvain community detection (after separating bi-connected components)",
-            4: "\n\tStrategy: "
-               "Community detection by cosine of squared adjacency matrix"
-               "(after separating bi-connected components)",
-            5: "\n\tStrategy: "
-               "Fast Community detection {partition + detect + merge}\n\t"
-               "(pipeline: bi-connected + partition + bi-connected + k-cliques + merge)",
-            6: "\n\tStrategy: "
-               "Community detection by extensive community-detection"
-               "\n\t(bi-connected + k3-clique + cosine similarity (with/out squaring)."
-        }
-        if self.args.strategy not in strategy_switcher:
-            exit("\033[93m Wrong input argument: --separation-strategy!\033[0m")
+        alg_list = self.args.strategy.split(".")
+        if not alg_list:
+            exit("\033[91m Input argument not provided: --separation-strategy!\033[0m")
+        if not set(alg_list).issubset(
+                {"bc", "k3", "cos", "sqCos", "sqcos", "louvain", "distributed"}):
+            exit("\033[91m Wrong input argument: --separation-strategy!\033[0m")
```

Comment:
Same suggestion as above, and also print the values in `alg_list` that are not in the whitelist.